### PR TITLE
CLDR-13969 Make ST faster by avoiding needless DOM manipulation

### PR DESCRIPTION
**Pull request by btangmu**
-New cldrChecksum to see whether row data has changed; skip DOM update if unchanged

-New unit tests for cldrChecksum

-Remove unused valueToItem

-Comments

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13969
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js [hunk 1]
```diff
@@ -325,19 +325,52 @@ const cldrSurveyTable = (function() {
 	 * AND by insertFixInfo in review.js (Dashboard "Fix")!
 	 */
 	function updateRow(tr, theRow) {
+		if (!tr || !theRow) {
+			return;
+		}
+		if (tr.checksum !== undefined) {
```
Comment (macchiati) [inline tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js:331]: minor.

Move const newChecksum = cldrChecksum(JSON.stringify(theRow)); above the undefined loop, call it "rowChecksum" (and change references below).

File: tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js [hunk 2]
```diff
@@ -325,19 +325,52 @@ const cldrSurveyTable = (function() {
...
+			const newChecksum = cldrChecksum(JSON.stringify(theRow));
+			if (newChecksum === tr.checksum) {
+				return;
+			}
+			tr.checksum = newChecksum;
```
Comment (macchiati) [inline tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js:336]: move this to outside the conditional

File: tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js [hunk 3]
```diff
@@ -325,19 +325,52 @@ const cldrSurveyTable = (function() {
...
+		} else {
```
Comment (macchiati) [inline tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js:337]: delete 337, 338

File: tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js (hunk 1 above)
Comment (btangmu) [reply tools/cldr-apps/WebContent/js/CldrSurveyVettingTable.js:331]: Good idea. I'll make another PR.

**Comment by btangmu** [comment]
I'm closing this. See https://github.com/unicode-org/cldr/pull/569 instead.

**Pull request closed without merging**